core/src/components/tab-button/test/shape/tab-button.e2e.ts

```diff
@@ -0,0 +1,51 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+import type { E2EPage, E2EPageOptions, ScreenshotFn } from '@utils/test/playwright';
+
+class TabButtonFixture {
+  readonly page: E2EPage;
+
+  constructor(page: E2EPage) {
+    this.page = page;
+  }
+
+  async goto(config: E2EPageOptions) {
+    const { page } = this;
+    await page.goto(`/src/components/tab-button/test/shape`, config);
+  }
+
+  async screenshot(screenshotModifier: string, screenshot: ScreenshotFn, buttonId: string) {
+    const { page } = this;
+
+    const screenshotString = screenshot(`tab-button-${screenshotModifier}`);
+    const tabButton = page.locator(buttonId);
+
+    await expect(tabButton).toHaveScreenshot(screenshotString);
+  }
+}
+
+/**
+ * This behavior does not vary across directions.
+ */
+configs({ modes: ['ionic-md'], directions: ['ltr'] }).forEach(({ config, screenshot, title }) => {
```

Comment:
All the screenshots are wrong. They do not match the HTML test page. Please fix.

Reply:
I have since fixed those screenshots. However, the basic tab-bar tests were also failing. On the one hand, it makes sense since the default button corners have changed. On the other hand, I do not understand how the border color could have changed:
![image](https://github.com/user-attachments/assets/b76583f6-283d-4250-a88a-e6777ccabb22)
Perhaps it's related to design tokens?

Reply:
I also just noticed that the snapshots for the "darwin" OS for the tab-button are also wrong but somehow the tests pass locally 😕

Reply:
`darwin` snapshots are from tests that were run on your local computer. Those snapshots may vary based on the computer, for example snapshots on a Windows device will most likely not be the same as a Mac device. So they're not considered the true source and are not allowed to be pushed to the repro.

In order to make sure [snapshots are always being generated correctly](https://github.com/ionic-team/ionic-framework/blob/000f55303e459c583e642337fb1894f419f37d48/docs/core/testing/usage-instructions.md#running-tests-inside-docker) to the same environment as the CI, we run the tests within Docker: `npm run test.e2e.docker`. You'll know that the snapshots are correct when they have the `linux` keyword within the file.

Make sure you are using that command to run the tests. You can also use `npm run test.e2e.docker.update-snapshots` to [update the true snapshots](https://github.com/ionic-team/ionic-framework/blob/000f55303e459c583e642337fb1894f419f37d48/docs/core/testing/usage-instructions.md#generating-or-updating-ground-truths-with-docker-local-development) locally.

In order to fix this PR, I suggest running:
1. `npm run test.e2e.docker tab-bar tab-button`. This command will run all the tests that have the keywords tab-bar or tab-bar.
2. Review the failing tests: `npx playwright show-report`
3. It's extremely important that you understand why the tests are failing. They should only be failing for `ionic` theme and anything related to the tab button radius.
4. If you feel confident, then update the snapshots: `npm run test.e2e.docker.update-snapshots tab-button tab-bar`. You might not need to update both components so adjust the command as needed.

Reply:
I see! That makes a more sense! Thanks for the context!
I noticed yesterday or the day before that I was running snapshot tests incorrectly and started using Docker like you mentioned. What I wasn't aware was that those "darwin" files were leftovers from running the tests incorrectly. I will remove them then.

By the way, did you get a chance to look at my first comment regarding the unexpected change in border color for the tab-button?

Reply:
I did, that's why I said to make sure you review the failing tests. I'm not seeing a change of color. The tests are failing for another reason.

Reply:
As I said before, the remaining tab-bar tests are failing due to the default border corners now being different. But, besides that, I also find it odd that the border color for the tab-button in the image I sent [here](https://github.com/ionic-team/ionic-framework/pull/30057#discussion_r1869223573) has also changed to a different shade of blue. As such, I am raising a flag before updating the snapshots that might be incorrect.

Reply:
After looking into a bit mor. I can indeed confirm that the color change was caused by the change in design tokens [here](https://github.com/ionic-team/ionic-framework/pull/29950. Thus, I conclude that snapshot tests do not check for color changes since these tests did not break with Bernardo's PR. I will then proceed with updating the snapshots